src/Compilers/CSharp/Portable/BoundTree/Constructors.cs

```diff
@@ -175,14 +175,14 @@ public static BoundCall Synthesized(SyntaxNode syntax, BoundExpression? receiver
             return Synthesized(syntax, receiverOpt, method, ImmutableArray.Create(arg0, arg1));
         }
 
-        public static BoundCall Synthesized(SyntaxNode syntax, BoundExpression? receiverOpt, MethodSymbol method, ImmutableArray<BoundExpression> arguments)
+        public static BoundCall Synthesized(SyntaxNode syntax, BoundExpression? receiverOpt, MethodSymbol method, ImmutableArray<BoundExpression> arguments, ImmutableArray<RefKind> argumentRefKindsOpt = default)
         {
             return new BoundCall(syntax,
                     receiverOpt,
                     method,
                     arguments,
                     argumentNamesOpt: default(ImmutableArray<string>),
-                    argumentRefKindsOpt: method.ParameterRefKinds,
+                    argumentRefKindsOpt: argumentRefKindsOpt.IsDefault ? method.ParameterRefKinds : argumentRefKindsOpt,
```

Comment:
> method.ParameterRefKinds [](http://example.com/codeflow?start=72&length=25)

Should probably adjust these as ```GetEffectiveArgumentRefKinds``` does that.

Reply:
There is also code that was just merged into 'main' that will need an adjustment as well - [https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs#L820C1-L838C1](https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs#L820C1-L838C1)

Reply:
> Should probably adjust these as `GetEffectiveArgumentRefKinds` does that.

I think each caller of this `BoundCall.Synthesized` should ensure arguments go through `LocalRewriter.MakeArguments`. But will add an assert here, good idea.

> There is also code that was just merged into 'main' that will need an adjustment as well

Thanks, will adjust when main flows here through [https://github.com/dotnet/roslyn/pull/68800](https://github.com/dotnet/roslyn/pull/68800).

Reply:
> I think each caller of this ```BoundCall.Synthesized ``` should ensure arguments go through ```LocalRewriter.MakeArguments```.

I do not think we can make an assumption like that. I see several call sites outside of ```LocalRewriter```. I think it will be more robust to build proper set of RefKinds, when caller doesn't supply any.